cmd/dep/ensure.go

```diff
@@ -163,6 +163,26 @@ func (cmd *ensureCommand) Run(ctx *dep.Ctx, args []string) error {
 	}
 
 	newLock := dep.LockFromSolution(solution)
+
+	// check if any projects are missing from vendor, if they are
+	// we'll need to write the vendor directory.
+	if writeV != dep.VendorAlways {
+		for _, proj := range newLock.Projects() {
+			projVendorPath := filepath.Join(p.AbsRoot, "vendor", string(proj.Ident().ProjectRoot))
+			projInVendor, err := fs.IsNonEmptyDir(projVendorPath)
+			if err != nil {
+				return errors.Wrapf(err, "ensure %v is a directory or remove it")
```

Comment:
Thanks @groob, very strange as I was switching between variables at the time, and seemed to have incorrectly settled on neither.

But I'm more concerned that this is even required, as the functionality seems to be missing (as per the issue), but even the README documents this should work. So I don't actually think this PR is even required, or at least the bug is likely elsewhere.